config/neptune/config.go

```diff
@@ -0,0 +1,206 @@
+
+	"github.com/crossplane-contrib/provider-jet-aws/config/common"
+)
+
+// Configure adds configurations for neptune group
+func Configure(p *config.Provider) {
+	p.AddResourceConfigurator("aws_neptune_cluster", func(r *config.Resource) {
+		r.Version = common.VersionV1Alpha2
+		r.ExternalName = config.ExternalName{
+			SetIdentifierArgumentFn: func(base map[string]interface{}, name string) {
+				base["cluster_identifier"] = name
+			},
+
+			OmittedFields: []string{
+				"cluster_identifier",
+				"cluster_identifier_prefix",
+			},
+
+			GetExternalNameFn: config.IDAsExternalName,
+			GetIDFn:           config.ExternalNameAsID,
+		}
+	})
+
+	p.AddResourceConfigurator("aws_neptune_cluster_endpoint", func(r *config.Resource) {
+		r.Version = common.VersionV1Alpha2
+		r.ExternalName = config.ExternalName{
+			SetIdentifierArgumentFn: func(base map[string]interface{}, name string) {
+				base["cluster_endpoint_identifier"] = name
+			},
+
+			OmittedFields: []string{
+				"cluster_endpoint_identifier",
+			},
+
+			GetExternalNameFn: func(tfstate map[string]interface{}) (string, error) {
+				id, ok := tfstate["id"].(string)
+				if !ok || id == "" {
+					return "", errors.New("cannot get id from tfstate")
+				}
+
+				// my_cluster:my_cluster_endpoint
+				w := strings.Split(id, ":")
+				if len(w) != 2 {
+					return "", errors.New("format of id should be my_cluster:my_cluster_endpoint")
+				}
+				return w[len(w)-1], nil
+			},
+
+			GetIDFn: func(_ context.Context, externalName string, parameters map[string]interface{}, _ map[string]interface{}) (string, error) {
+				ci, ok := parameters["cluster_identifier"].(string)
+				if !ok || ci == "" {
+					return "", errors.New("cannot get cluster_identifier from parameters")
+				}
+				return fmt.Sprintf("%s:%s", ci, externalName), nil
+			},
+		}
+
+		r.References["cluster_identifier"] = config.Reference{
+			Type: "Cluster",
+		}
+	})
+
+	p.AddResourceConfigurator("aws_neptune_cluster_instance", func(r *config.Resource) {
+		r.Version = common.VersionV1Alpha2
+		r.ExternalName = config.ExternalName{
+			SetIdentifierArgumentFn: func(base map[string]interface{}, name string) {
+				base["identifier"] = name
+			},
+
+			OmittedFields: []string{
+				"identifier",
+				"identifier_prefix",
+			},
+
+			GetExternalNameFn: config.IDAsExternalName,
+			GetIDFn:           config.ExternalNameAsID,
+		}
+
+		r.UseAsync = true
+
+		r.References["cluster_identifier"] = config.Reference{
+			Type: "Cluster",
+		}
+	})
+
+	p.AddResourceConfigurator("aws_neptune_cluster_parameter_group", func(r *config.Resource) {
+		r.Version = common.VersionV1Alpha2
+		r.ExternalName = config.ExternalName{
+			SetIdentifierArgumentFn: func(base map[string]interface{}, name string) {
+				base["name"] = name
+			},
+
+			OmittedFields: []string{
+				"name",
+				"name_prefix",
+			},
+
+			GetExternalNameFn: config.IDAsExternalName,
+			GetIDFn:           config.ExternalNameAsID,
+		}
+	})
```

Comment:
When we remove these configurations, following part is generated in the `zz_subnetgroup_types.go` file:
```
	// +kubebuilder:validation:Optional
	Name *string `json:"name,omitempty" tf:"name,omitempty"`
```
Is this something we want? Won't this cause issues when we set this value different than the CR name?

Reply:
> When we remove these configurations, following part is generated in the zz_subnetgroup_types.go file:

Are you removing them or replacing with the `NameAsExternalName` configuration from Terrajet? Because it's exactly the same as the configuration you wrote here so I'd not expect anything to change in the generated code actually.